### ISSUE TITLE
S3Proxy does not delete Swift MPU parts

### ISSUE THREAD

**Issue by andrewgaul** [andrewgaul/s3proxy#92]
S3Proxy needs to add [?multipart-manifest=delete](http://docs.openstack.org/developer/swift/overview_large_objects.html#deleting-a-large-object) to its Swift remove object call to delete all parts of an MPU.  Need to think about this a bit since we would send this unconditionally for all `removeBlob` calls.

```
$ s3cmd -c .s3cfg-s3proxy put --multipart-chunk-size-mb=5 work/s3proxy/target/s3proxy s3://whatever
asking for bucket location
work/s3proxy/target/s3proxy -> s3://whatever/s3proxy  [part 1 of 2, 5MB]
 5242880 of 5242880   100% in    7s   647.26 kB/s  done
work/s3proxy/target/s3proxy -> s3://whatever/s3proxy  [part 2 of 2, 2MB]
 2508615 of 2508615   100% in    4s   593.93 kB/s  done

$ s3cmd -c .s3cfg-s3proxy ls --recursive s3://whatever
2015-11-13 20:30   7751495   s3://whatever/s3proxy
2015-11-13 20:30   5242880   s3://whatever/s3proxy/slo/1447446602.394000/0/0/00000001
2015-11-13 20:30   2508615   s3://whatever/s3proxy/slo/1447446602.394000/0/0/00000002

$ s3cmd -c .s3cfg-s3proxy rm s3://whatever/s3proxy
File s3://whatever/s3proxy deleted

$ s3cmd -c .s3cfg-s3proxy ls --recursive s3://whatever
2015-11-13 20:30   5242880   s3://whatever/s3proxy/slo/1447446602.394000/0/0/00000001
2015-11-13 20:30   2508615   s3://whatever/s3proxy/slo/1447446602.394000/0/0/00000002
```

Found by @dleute in #91.

**Comment by dleute** [comment]
@andrewgaul The change to fix this now prevents regular files from being deleted.

11:04:28.854 [S3Proxy-22 - /testbucket/s3proxy] DEBUG jclouds.wire - << "{"Number Not Found": 0, "Response Status": "400 Bad Request", "Errors": [["/testbucket/s3proxy", "Not an SLO manifest"]], "Number Deleted": 0, "Response Body": ""}"

Should we re-open this issue or  create a new one?

**Comment by andrewgaul** [comment]
@dleute Which Swift server are you using?  I did not see these symptoms when using Rackspace Cloud Files.

**Comment by dleute** [comment]
@andrewgaul I'm on softlayer in dal05 for this test.

**Comment by dleute** [comment]
FYI: no error is reported by the CLI. You have to watch the logs of s3proxy to see the error. And the file is never actually deleted.

That's been another issue. Errors have sometimes been hidden by s3proxy/jclouds. Shouldn't this have become a failed delete?

**Comment by andrewgaul** [comment]
@dleute Thanks for the additional information!  I reproduced your symptoms and reverted the change in upstream jclouds.  You are correct that jclouds should report some kind of error for this and I believe it does not because remove tends to ignore errors like 404 file not found.  For something like 400 bad request it should raise this.  I am traveling right now and not available for further investigation but will look into this further when I return.  In the mean time I will reopen this issue.

**Comment by dleute** [comment]
@andrewgaul Thanks!

**Comment by dleute** [comment]
Thanks!

**Comment by dleute** [comment]
FYI: I'm still getting not an SLO manifest. Is it possible the change hasn't propagated yet?

**Comment by dleute** [comment]
At the very least, jclouds is still sending multipart-manifest=delete

**Comment by andrewgaul** [comment]
I checked https://repository.apache.org/content/groups/snapshots/org/apache/jclouds/api/openstack-swift/2.0.0-SNAPSHOT/openstack-swift-2.0.0-20151208.052256-460-sources.jar and see that `removeBlob` no longer calls `getStaticLargeObjectApi`.  Can you rebuild S3Proxy via `mvn package --update-snapshots` to see if this helps?

**Comment by dleute** [comment]
Thanks! I think my IDE didn't catch the changes and needed to be refreshed as well.

**Comment by andrewgaul** [comment]
Another attempt at a fix: jclouds/jclouds#909.
